src/ServerScriptService/InstanceComponents/Structures/Observer.lua

```diff
@@ -0,0 +1,38 @@
+local Players = game:GetService("Players")
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local RunService = game:GetService("RunService")
+
+local Settings = require(ReplicatedStorage.Common.Config.Terrain)
+local MinimumRange = Settings.MinimumRange
+
+local Observer = {}
+Observer.__index = Observer
+
+function Observer.new(instance)
+	local self = setmetatable({}, Observer)
+	self.Instance = instance
+
+    print(self.Instance.Name, " created!")
+
+    RunService.Heartbeat:Connect(function()
+        for _,v in pairs(Players:GetPlayers()) do
+            if not v.Character then return end
+            local distanceBetweenParts = (v.Character.HumanoidRootPart.Position - self.Instance.PrimaryPart.Position).Magnitude
+            if distanceBetweenParts > MinimumRange then
+                self.Instance.PrimaryPart.Color = Color3.fromRGB(0,255,0)
+            else
+                self.Instance.PrimaryPart.Color = Color3.fromRGB(255,0,0)
+            end
+        end
+    end)
+
+	return self
```

Comment:
Many weird things have happened with your indentation here. Sometimes this is caused by tab/space mismatch. If it looks normal in your editor, run 'convert tabs to spaces' from the command bar - That usually resolves it for most people.